lib/Request.js

```diff
@@ -143,16 +143,23 @@ Request.prototype.propertyError = function(property, errorCode, httpStatusCode,
     .then(function(options) {
         if (options.schema.properties.hasOwnProperty(property)) {
             // Get details about the error code
-            var error = options.schema.properties[property].errorCodes.filter(function(item) {
-                return item.code == errorCode;
-            })[0];
-
-            properties.push({
-                property: property,
-                code: error.code,
-                error: error.error,
-                message: error.message
-            });
+            if (errorCode >= 1000) {
+                var error = options.schema.properties[property].errorCodes.filter(function (item) {
```

Comment:
by returning an "error" field in the validation schema below we don't have to do this differntiator anymore